### ISSUE TITLE
Redux everything

### ISSUE THREAD

**Issue by marrus-sh** [marrus-sh/mastodon-go#1]
Here's the list:

- [x] `account`
    - [x] `GET /api/v1/accounts/:id`
    - [x] `GET /api/v1/accounts/verify_credentials`
        - Not presently needed but a requirement for remote login support
    - [x] `PATCH /api/v1/accounts/update_credentials`
        - Not blocking but would be nice to have
- [x] `attachment`
    - [x] `POST /api/v1/media`
- [x] `card`
    - [x] `GET /api/v1/statuses/:id/card`
- [x] `catalogue`
    - [x] `GET /api/v1/accounts/:id/followers`
    - [x] `GET /api/v1/accounts/:id/following`
    - [ ] `GET /api/v1/accounts/search`
        - Probably won't support
    - [x] `GET /api/v1/blocks`
    - [x] `GET /api/v1/follow_requests`
    - [x] `GET /api/v1/mutes`
    - [x] `GET /api/v1/statuses/:id/reblogged_by`
    - [x] `GET /api/v1/statuses/:id/favourited_by`
- [x] `conversation`
    - [x] `GET /api/v1/statuses/:id/context`
- [x] `courier`
    - [x] `GET /api/v1/notifications`
- [x] `directory`
    - [x] `GET /api/v1/domain_blocks`
- [x] `domain`
    - [x] `POST /api/v1/domain_blocks`
    - [x] `DELETE /api/v1/domain_blocks`
- [x] `emoji`
    - [x] `GET /api/v1/custom_emojis`
- [ ] `instance`
    - [ ] `GET /api/v1/instance`
        - Not implementing… *yet* (it'd be nice if this info was sent via `#initial-state` tho)
- [x] `meta`
    - [x] `#initial-state`
- [x] `notification`
    - [x] `GET /api/v1/notifications/:id`
    - [x] `POST /api/v1/notifications/clear`
    - [x] `POST /api/v1/notifications/dismiss`
- [x] `relationship`
    - [x] `POST /api/v1/accounts/:id/follow`
    - [x] `POST /api/v1/accounts/:id/unfollow`
    - [x] `POST /api/v1/accounts/:id/block`
    - [x] `POST /api/v1/accounts/:id/unblock`
    - [x] `POST /api/v1/accounts/:id/mute`
    - [x] `POST /api/v1/accounts/:id/unmute`
    - [x] `GET /api/v1/accounts/relationships`
    - [x] `POST /api/v1/follow_requests/:id/authorize`
    - [x] `POST /api/v1/follow_requests/:id/reject`
    - [ ] `POST /api/v1/follows`
        - Probably won't be supported
- [ ] `report`
    - [ ] `GET /api/v1/reports`
    - [ ] `POST /api/v1/reports`
- [ ] `search`
    - [ ] `GET /api/v1/search`
- [ ] `setting`
    - [x] `localStorage()`
- [x] `status`
    - [x] `GET /api/v1/statuses/:id`
    - [x] `POST /api/v1/statuses`
    - [x] `DELETE /api/v1/statuses/:id`
    - [x] `POST /api/v1/statuses/:id/reblog`
    - [x] `POST /api/v1/statuses/:id/unreblog`
    - [x] `POST /api/v1/statuses/:id/favourite`
    - [x] `POST /api/v1/statuses/:id/unfavourite`
    - [x] `POST /api/v1/statuses/:id/mute`
    - [x] `POST /api/v1/statuses/:id/unmute`
- [x] `timeline`
    - [x] `GET /api/v1/accounts/:id/statuses`
    - [x] `GET /api/v1/favourites`
    - [x] `GET /api/v1/timelines/home`
    - [x] `GET /api/v1/timelines/public`
    - [x] `GET /api/v1/timelines/tag/:hashtag`
